From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct scm_creds
Patch-mainline: never, kabi

In 3.12.57, commit 415e3d3e90ce9e18727e8843ae343eda5a58fad6 (unix:
correctly track in-flight fds in sending process user_struct),
upstream commit 415e3d3e90ce9e18727e8843ae343eda5a58fad6 added one
member to struct scm_creds. It indeed changed the layout of that
structure.

But given this structure is internal to scm and allocated there, we
can just protect the change the usual way, by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/scm.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/net/scm.h
+++ b/include/net/scm.h
@@ -21,7 +21,9 @@ struct scm_creds {
 struct scm_fp_list {
 	short			count;
 	short			max;
+#ifndef __GENKSYMS__
 	struct user_struct	*user;
+#endif
 	struct file		*fp[SCM_MAX_FD];
 };
 
